shell/browser/api/electron_api_session.cc

```diff
@@ -890,6 +934,71 @@ void Session::OnExtensionReady(content::BrowserContext* browser_context,
                                const extensions::Extension* extension) {
   Emit("extension-ready", extension);
 }
+
+void Session::SetChromeAPIHandlers(const gin_helper::Dictionary& api,
+                                   gin::Arguments* args) {
+  GetTabHandler get_tab;
+  GetActiveTabHandler get_active_tab;
+
+  if (api.Get("getTab", &get_tab)) {
+    get_tab_handler_ = std::make_unique<GetTabHandler>(get_tab);
+  }
+
+  if (api.Get("getActiveTab", &get_active_tab)) {
+    get_active_tab_handler_ =
+        std::make_unique<GetActiveTabHandler>(get_active_tab);
+  }
+}
+
+std::unique_ptr<ChromeTabDetails> Session::GetChromeTabDetails(
+    WebContents* tab_contents) {
+  v8::Isolate* isolate = JavascriptEnvironment::GetIsolate();
+  v8::HandleScope scope(isolate);
+
+  ChromeTabDetails details;
+
+  if (!get_tab_handler_) {
+    if (!has_emitted_chrome_tabs_get_warning_) {
+      EmitWarning(
+          node::Environment::GetCurrent(isolate),
+          "No handler has been registered for `chrome.tabs.get`. Please use "
+          "`session.setChromeAPIHandlers`.",
+          "electron");
+      has_emitted_chrome_tabs_get_warning_ = true;
+    }
+
+    return std::make_unique<ChromeTabDetails>(details);
```

Comment:
🤔  should we fill this out with `ExtensionTabUtil::CreateTabObject` by default?

Reply:
This method is only responsible for calling the callback specified by the user and is needed to merge the user specified data in `CreateTabObject` which also scrubs relevant data based on `Extension`.